### PR TITLE
[serve] Create all other actors in master actor

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Another step towards a more centralized architecture where the master actor manages all other actors in the system - creating them directly in the master actor instead of using the `start_actor_with_creator` API. Greatly simplifies the `GlobalState` logic.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by edoakes** [comment]
Dependent on https://github.com/ray-project/ray/pull/7774

**Comment by simon-mo** [comment]
Can you resolve the conflict? I just merged  #7774

**Comment by edoakes** [comment]
@simon-mo fixed conflicts and updated this to also create the backend replicas in the master actor.

File: python/ray/serve/api.py [hunk 1]
```diff
@@ -114,6 +114,10 @@ def init(
     if not ray.is_initialized():
         ray.init(**ray_init_kwargs)
 
+    # Register serialization context once
+    ray.register_custom_serializer(Query, Query.ray_serialize,
```
Comment (simon-mo) [inline python/ray/serve/api.py]: this is already done in line 130

File: python/ray/serve/api.py [hunk 2]
```diff
@@ -138,16 +142,15 @@ def init(
     def kv_store_connector(namespace):
         return SQLiteKVStore(namespace, db_path=kv_store_path)
 
-    master = start_initial_state(kv_store_connector)
+    master = ServeMaster.remote(kv_store_connector)
```
Comment (simon-mo) [inline python/ray/serve/api.py]: `ServeMaster.options(name= SERVE_MASTER_NAME).remote(kv_store_connector)`

File: python/ray/serve/global_state.py [hunk 3]
```diff
@@ -88,83 +128,24 @@ class GlobalState:
...
+        self.master_actor = master_actor
 
-        # Connect to all the table
-        bootstrap_config = ray.get(
-            self.master_actor_handle.get_bootstrap_state_dict.remote())
-        kv_store_connector = bootstrap_config[BOOTSTRAP_KV_STORE_CONN_KEY]
+        # Connect to all the tables.
+        kv_store_connector = ray.get(
+            self.master_actor.get_kv_store_connector.remote())
         self.route_table = RoutingTable(kv_store_connector)
```
Comment (simon-mo) [inline python/ray/serve/global_state.py]: do we still need the tables in GlobalState?

**Review by simon-mo — APPROVED** [review]
LGTM nice refactoring. There will be a follow up PR moving tables access to master? e.g. in _scale, we check for 
```python
assert (backend_tag in global_state.backend_table.list_backends()), "Backend {} is not registered.".format(backend_tag)
```

**Comment by edoakes** [comment]
@simon-mo yes, see:
https://github.com/ray-project/ray/pull/7835

File: python/ray/serve/api.py (hunk 1 above)
Comment (edoakes) [reply python/ray/serve/api.py]: Oh, bad merge. Fixed.

File: python/ray/serve/global_state.py (hunk 3 above)
Comment (edoakes) [reply python/ray/serve/global_state.py]: They'll be removed in follow-up

**Pull request merged**